azerothcore/azerothcore-wotlk
# fix(Core/Groups): fix group enchanting level not resetting properly

## Changes Proposed:
-  Call `ResetMaxEnchantingLevel()` after removing the member from the party slot, so that the computed enchanting level takes into account the member leaving


## Issues Addressed:
- Closes https://github.com/chromiecraft/chromiecraft/issues/96



## Tests Performed:

- Applied the C++ fix
- Tested in-game on Windows 10 clients and server


## How to Test the Changes:

Setup:
Get three characters
- Character A: without the enchanting skill
- Character B: without the enchanting skill
- Character C: **with** the enchanting profession

First step:
- Get characters A, B and C into a group
- Move all characters to Dr. Whitherlimb (`.go c 1975927`)
- Kill Dr. Whitherlimb
- He should loot a green item, verify that you are able to disenchant it:
![image](https://user-images.githubusercontent.com/24357633/116296339-65397a80-a79a-11eb-87e1-2456311a7280.png)

Second step:
- Remove all three characters from the group
- Get characters A, B and C into a group
- With character C, leave the group
- Do `.respawn` on the Dr.
- Kill Dr. Whitherlimb
- He should loot a green item, verify that you are **NOT** able to disenchant it:
![image](https://user-images.githubusercontent.com/24357633/116296571-acc00680-a79a-11eb-8cdc-d1442fa31b16.png)

Third step: (thanks @eTzmNcbkrng for pointing this out)
- Remove all characters from the group
- Get characters A, B and C into a group
- Disconnect character C from the game without removing them from the group (this can be achieved for example by using the `.kick` command in the server console):
![image](https://user-images.githubusercontent.com/24357633/116466255-8cfa1280-a86e-11eb-8d34-9a9e7ac855de.png)
- Do `.respawn` on the Dr.
- Kill Dr. Whitherlimb
- He should loot a green item, verify that you are **NOT** able to disenchant it:
![image](https://user-images.githubusercontent.com/24357633/116466205-7ce23300-a86e-11eb-9697-bfb5b7c325bf.png)


(if this is not clear, see my comment and video on how to reproduce the bug on the issue: https://github.com/chromiecraft/chromiecraft/issues/96#issuecomment-827807897)

## Known Issues and TODO List:

None


## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
